src/test/java/com/openshift/internal/restclient/model/v1/RouteTest.java

```diff
@@ -63,7 +64,42 @@ public void createTLSConfigWhenUndefined() {
 		assertEquals("", tls.getKey());
 	}
 	
-	@Test
+    @Test
+    public void getPortWhenUndefined() {
+        ModelNode node = ModelNode.fromJSONString(Samples.V1_ROUTE_WO_TLS.getContentAsString());
+        route = new Route(node, client, ResourcePropertiesRegistry.getInstance().get(VERSION, ResourceKind.ROUTE));
+        ITargetPort port = route.getPort();
+        assertNull(port);
```

Comment:
I see a Route#createPort(). If nothing further is done (which is most likely in user code, not included in this PR) I see a TargetPort being create that will return -1 if you dont set it a numeric port. Missing something?